In [1]:
import pandas as pd

https://www.kaggle.com/datasets/sohaibdevv/consumer-shopping-behavior-and-preference-study-2026

In [2]:
compras = pd.read_csv('compras_online_store.csv')

In [5]:
compras.shape

(11789, 25)

In [6]:
compras.head()

,age,monthly_income,daily_internet_hours,smartphone_usage_years,social_media_hours,online_payment_trust_score,tech_savvy_score,monthly_online_orders,monthly_store_visits,avg_online_spend,...,free_return_importance,product_availability_online,impulse_buying_score,need_touch_feel_score,brand_loyalty_score,environmental_awareness,time_pressure_level,gender,city_tier,shopping_preference
0,56,221111,6.5,12,0.7,1,6,16,16,28551,...,7,7,1,3,6,5,2,Other,Tier 3,Store
1,69,96029,8.2,13,2.7,6,9,14,1,124056,...,3,4,9,6,8,1,7,Male,Tier 3,Hybrid
2,46,19055,6.4,4,2.1,10,8,2,0,81939,...,4,10,1,1,3,3,3,Female,Tier 3,Store
3,32,53170,6.4,11,0.7,2,10,20,3,35901,...,10,2,4,8,2,6,6,Female,Tier 1,Store
4,60,244016,6.0,5,0.7,2,5,18,16,131971,...,2,5,8,9,7,1,6,Male,Tier 3,Store


O arquivo trata-se de um conjunto de informações sobre compras online e em lojas fisicas, para um estudo comprativo entre as duas opções de compras.
Ele possui 11789 linhas e 25 colunas.
As colunas são: idade, renda mensal, horas diárias na internet, anos de uso do smartphone, horas nas redes sociais, nivel confiança em pagamento online, nível conhecimento tecnológico, pedidos online mensais, visitas lojas mensais, gasto médio online, importância devolução grátis, disponibilidade produto online, nível compra por impulso, nível necessidade de tocar sentir(avaliar fisicamente), nível lealdade marca, conscientização ambiental, nivel pressão tempo, genero, porte da cidade, preferências compras.
O data set não possui dados nulos e nem valores constantes.
O dataset apresentada as seguintes variáveis: Interger, Float, Strings.
Das 25 colunas, 20 são do tipo int, 2 flot e 3 string.


Perguntas:
1. O nível de consciencia ambiental aumenta ou diminui de acordo com a idade e renda?
2. A quantidade de anos de uso do smartphone influencia na quantidade de compras online?
3. Quem passa mais horas nas redes sociais tem um gasto maior em lojas fisicas ou online?
4. o nível de cohecimento tecnológico interfere no nível de confiança nos meios de pagamento online?.
5. Qual a média de idade das pessoas que tem o maior volume de compras online?
6. Qual a preferencia de canal de compra de acordo com o porte da cidade
7. Qual a preferencia de canal de compras de acordo com o  genero?
8. Quem tem maior nível de compras por impulso prefere comprar online ou fisicamente
9. Relação da renda com o canal preferido de compras
10. Qual a média de idade das pessoas que tem o maior volume de compras fisicas?


In [7]:
# Nível de confiança em pagamentos online vai de 1 a 10
compras['online_payment_trust_score'].unique()

array([ 1,  6, 10,  2,  4,  5,  8,  7,  9,  3])

In [8]:
# Nível de conhecimento tecnológico vai de 1 a 10
compras['tech_savvy_score'].unique()

array([ 6,  9,  8, 10,  5,  7,  3,  1,  4,  2])

In [9]:
# Nível de importância de devolução grátis vai de 1 a 10
compras['free_return_importance'].unique()

array([ 7,  3,  4, 10,  2,  9,  6,  1,  5,  8])

In [10]:
# Nível de compra por impulso 1 a 10
compras['impulse_buying_score'].unique()

array([ 1,  9,  4,  8,  6, 10,  5,  7,  2,  3])

In [11]:
# Tipos de genero Masculino, feminino e outros
compras['gender'].unique()

<ArrowStringArray>
['Other', 'Male', 'Female']
Length: 3, dtype: str

In [12]:
# Porte da cidade 1,2 e 3
compras['city_tier'].unique()

<ArrowStringArray>
['Tier 3', 'Tier 1', 'Tier 2']
Length: 3, dtype: str

In [13]:
#Tipos de preferencia de compras Loja, Hibrido e Online
compras['shopping_preference'].unique()

<ArrowStringArray>
['Store', 'Hybrid', 'Online']
Length: 3, dtype: str

In [6]:
compras.info()

<class 'pandas.DataFrame'>
RangeIndex: 11789 entries, 0 to 11788
Data columns (total 25 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   age                          11789 non-null  int64  
 1   monthly_income               11789 non-null  int64  
 2   daily_internet_hours         11789 non-null  float64
 3   smartphone_usage_years       11789 non-null  int64  
 4   social_media_hours           11789 non-null  float64
 5   online_payment_trust_score   11789 non-null  int64  
 6   tech_savvy_score             11789 non-null  int64  
 7   monthly_online_orders        11789 non-null  int64  
 8   monthly_store_visits         11789 non-null  int64  
 9   avg_online_spend             11789 non-null  int64  
 10  avg_store_spend              11789 non-null  int64  
 11  discount_sensitivity         11789 non-null  int64  
 12  return_frequency             11789 non-null  int64  
 13  avg_delivery_days          

In [16]:
# 1. O nível de consciencia ambiental aumenta ou diminui de acordo com a idade e renda?

#verificando a faixa etária do dataset
compras['age'].sort_values().unique()

#definindo 3 faixas de idade 
# Faixa 1 - 18 a 35 anos
# Faixa 2 - 36 a 55 anos
# Faixa 3 - 56 a 79 anos
faixa1 = compras['age'] <= 35
print(faixa1.sum())


3364


In [ ]:
# 2. A quantidade de anos de uso do smartphone influencia na quantidade de compras online?


In [ ]:
# 3. Quem passa mais horas nas redes sociais tem um gasto maior em lojas fisicas ou online?


In [ ]:
# 4. o nível de cohecimento tecnológico interfere no nível de confiança nos meios de pagamento online?.


In [ ]:
# 5. Qual a média de idade das pessoas que tem o maior volume de compras online?


In [ ]:
# 6. Qual a preferencia de canal de compra de acordo com o porte da cidade


In [ ]:
# 7. Qual a preferencia de canal de compras de acordo com o  genero?


In [ ]:
# 8. Quem tem maior nível de compras por impulso prefere comprar online ou fisicamente


In [ ]:
# 9. Relação da renda com o canal preferido de compras


In [ ]:
# 10. Qual a média de idade das pessoas que tem o maior volume de compras fisicas?